src/extension.ts

```diff
@@ -136,6 +137,45 @@ export async function activate(context: vscode.ExtensionContext) {
   console.log('Clang Language Server is now active!');
   fileStatus.activate(client, context);
   switchSourceHeader.activate(client, context);
+  async function handleConfigFilesChanged(uri: vscode.Uri) {
+    if ((await vscode.workspace.fs.readFile(uri)).length) {
+      let relativePath = vscode.workspace.asRelativePath(uri, false);
+      // If it is at the top of the workspace
+      if (relativePath == path.basename(uri.fsPath)) {
```

Comment:
Why only listen to changes in the configuration file in the top of the workspace?
Currently clangd has been able to automatically find the compilation database located in the `build` folder.[D78629](https://reviews.llvm.org/D78629)

Reply:
You're making fair points, however I'm not sure those possibilities are that much used, so I don't know if they're worth the false positives we could get by removing this check (for example by having multiple build directories: if i compile my code in 3 different folders I don't want to have three messages popping up asking if I want to restart clang server). 

Of course my opinion is a bit biased (the example I made is *exactly* why I implemented that check), so before removing/keeping it I'd rather listen to the maintainer opinion

Reply:
I agree with @lh123 if we want to have a watch put on the compilation database this should be set by reading the parameter that is sent to `clangd`. There are numerous environments where `compile_commands.json` resides in other directories other than the top director or the build directory. For example @mozilla we use dedicated `obj` directory that can be named whatever you like and the `compile_commands.json` can be created in that directory.

Reply:
Hi and sorry for the delay. From your comments I see that i might have underestimated the utility of that option... I'll try to push some new commits in the next days (right now I do not have my pc near me)